api/prober/types.go

```diff
@@ -0,0 +1,67 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prober
+
+import (
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Config provides typed access to prober configuration
+type Config struct {
+	// InternalKubeConfigSecretName is the name of the kubernetes secret which has the kubeconfig to connect to the shoot control plane API server via internal domain
+	InternalKubeConfigSecretName string `json:"internalKubeConfigSecretName"`
+	// ExternalKubeConfigSecretName is the name of the kubernetes secret which has the kubeconfig to connect to the shoot control plane API server via external domain
+	ExternalKubeConfigSecretName string `json:"externalKubeConfigSecretName"`
+	// ProbeInterval is the interval with which the probe will be run
+	ProbeInterval *metav1.Duration `json:"probeInterval,omitempty"`
+	// InitialDelay is the initial delay in running a probe for the first time
+	InitialDelay *metav1.Duration `json:"initialDelay,omitempty"`
+	// ProbeTimeout is the timeout that is set on the client which is used to reach the shoot control plane API server
+	ProbeTimeout *metav1.Duration `json:"probeTimeout,omitempty"`
+	// SuccessThreshold is the number of consecutive times a probe is successful to ascertain that the probe is healthy
+	SuccessThreshold *int `json:"successThreshold,omitempty"`
+	// FailureThreshold is the number of consecutive times a probe is unsuccessful to ascertain that the probe is unhealthy
+	FailureThreshold *int `json:"failureThreshold,omitempty"`
+	// InternalProbeFailureBackoffDuration backoff duration if the internal probe is unhealthy, before attempting again
+	InternalProbeFailureBackoffDuration *metav1.Duration `json:"internalProbeFailureBackoffDuration,omitempty"`
+	// BackoffJitterFactor jitter with which a probe is run
+	BackoffJitterFactor *float64 `json:"backoffJitterFactor,omitempty"`
+	// DependentResourceInfos dependent resources that should be considered for scaling in case the shoot control API server cannot be reached via external domain
+	DependentResourceInfos []DependentResourceInfo `json:"dependentResourceInfos"`
+}
+
+// DependentResourceInfo captures a dependent resource which should be scaled
+type DependentResourceInfo struct {
+	// Ref identifies a resource
+	Ref *autoscalingv1.CrossVersionObjectReference `json:"ref"`
+	// ShouldExist should be true if this resource should be present. If the resource is optional then it should be false.
+	ShouldExist *bool `json:"shouldExist"`
```

Comment:
Can we change this field name to `Optional`, which is in line with many other Kubernetes resource definitions? In which case, the code needs to be changed to reflect this flipping of boolean role here.

Reply:
Done.